primitives/src/merkle_tree/append_only.rs

```diff
@@ -247,4 +266,20 @@ mod mt_tests {
             bincode::deserialize(&bincode::serialize(node).unwrap()).unwrap()
         );
     }
+
+    #[test]
+    fn test_mt_iter() {
+        test_mt_iter_helper::<Fq254>();
+        test_mt_iter_helper::<Fq377>();
+        test_mt_iter_helper::<Fq381>();
+    }
+
+    fn test_mt_iter_helper<F: RescueParameter>() {
+        let mt = RescueMerkleTree::<F>::from_elems(
+            Some(2),
+            &[F::from(0u64), F::from(1u64), F::from(2u64)],
+        )
+        .unwrap();
+        assert!(mt.iter().all(|(index, elem)| { elem == &F::from(*index) }));
```

Comment:
this test is slightly too hardcoded? can we maybe have 10 test for a randomly generated list of some size (randomly ranged from [10, 50], sth like that), 

also should we add test for MT that contains empty subtree, or forgotten subtree, and showcase what our iterator will do?

Reply:
IMHO we don't need such random test as the code doesn't manipulate the value. But sure I'll add a test for forgotten subtree.

Reply:
strengthened test [bd2bc19](https://github.com/EspressoSystems/jellyfish/pull/475/commits/bd2bc19dcc34a7b0647794c15cc820ff0fa27c79)